tripy/examples/resent50/tripy-res50.ipynb

```diff
@@ -0,0 +1,599 @@
+    "        ]\n",
+    "\n",
+    "        self.activation = tp.relu\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        identity = self.shortcut(x)\n",
+    "        for layer in self.layer:\n",
+    "            x = layer(x)\n",
+    "        x = x + identity\n",
+    "        x = self.activation(x)\n",
+    "        return x\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define resenet stage and encoder for the backbone model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ResNetStage(tp.Module):\n",
+    "    def __init__(self, num_layers, in_channels, out_channels, bottleneck_channels, stride):\n",
+    "        super(ResNetStage, self).__init__()\n",
+    "        self.layers = []\n",
+    "        for i in range(num_layers):\n",
+    "            layer = ResNetBottleNeckLayer(\n",
+    "                in_channels if i == 0 else out_channels,\n",
+    "                out_channels, bottleneck_channels,\n",
+    "                stride if i == 0 else (1, 1)\n",
+    "            )\n",
+    "            self.layers.append(layer)\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        for layer in self.layers:\n",
+    "            x = layer(x)\n",
+    "        return x\n",
+    "\n",
+    "\n",
+    "class ResNetEncoder(tp.Module):\n",
+    "    def __init__(self, layers_config):\n",
+    "        super(ResNetEncoder, self).__init__()\n",
+    "        self.stages = []\n",
+    "        in_channels = 64\n",
+    "        for idx, (num_layers, out_channels, bottleneck_channels, stride) in enumerate(layers_config):\n",
+    "            stage = ResNetStage(num_layers, in_channels, out_channels, bottleneck_channels, stride)\n",
+    "            self.stages.append(stage)\n",
+    "            in_channels = out_channels\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        for stage in self.stages:\n",
+    "            x = stage(x)\n",
+    "        return x\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define a max pooling class and embeddings class for input pre-procDefine a max pooling class and embeddings class for input pre-processing before feeding to encoderessing before feeding to encoder"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MaxPool2d(tp.Module):\n",
+    "    def __init__(self, kernel_dims, stride=None, padding=((0, 0), (0, 0))):\n",
+    "        super(MaxPool2d, self).__init__()\n",
+    "        self.kernel_dims = kernel_dims\n",
+    "        self.stride = stride if stride is not None else kernel_dims\n",
+    "        self.padding = padding\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        return tp.maxpool(x, kernel_dims=self.kernel_dims, stride=self.stride, padding=self.padding)\n",
```

Comment:
Can we use `tp.maxpool` directly instead of wrapping it in a module?

Reply:
done, removed module